tests/integration/awslambda/test_lambda_api.py

```diff
@@ -2412,7 +2434,15 @@ def test_lambda_provisioned_lifecycle(self, lambda_client, create_lambda_functio
 
 @pytest.mark.skip_snapshot_verify(
     condition=is_old_provider,
-    paths=["$..RevisionId", "$..Policy.Statement", "$..PolicyName", "$..PolicyArn", "$..Layers"],
+    paths=[
+        "$..RevisionId",
+        "$..Policy.Statement",
+        "$..PolicyName",
+        "$..PolicyArn",
+        "$..Layers",
+        # mismatching resource index due to SnapStart
+        "$..Statement.Condition.ArnLike.'AWS:SourceArn'",
```

Comment:
`:` break the JsonPath snapshot comparison and need to be enquoted using single quotes

Reply:
Maybe we should add some validation for this to avoid errors like this. Was this fairly obvious when debugging?

Reply:
The error was pretty specific (see below) but the stack trace wasn't particularly helpful in identifying the issue because the hook is executed after the test.
It could be nice to give some context when raising (e.g., which snapshot match or which verify skip path).

```
>       raise JsonPathParserError('Parse error at %s:%s near token %s (%s)'
                                  % (t.lineno, t.col, t.value, t.type))
E       jsonpath_ng.exceptions.JsonPathParserError: Parse error at 1:34 near token : (:)

.venv/lib/python3.10/site-packages/jsonpath_ng/parser.py:81: JsonPathParserError
```